### PR TITLE
support: --debug flag will print http requests for easier debugging

### PR DESCRIPTION
**Pull request by vadmeste**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license] (https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
This PR is not tested yet; this requires the help of @anjalshireesh to set up a local environment

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fmc%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by shtripat** [comment]
It shows details as below in a local deployment
```
---------START-HTTP---------
HEAD / HTTP/1.1
Host: localhost:9000
User-Agent: Go-http-client/1.1
Content-Type: application/json

HTTP/1.1 200 OK
Content-Length: 3035
Accept-Ranges: bytes
Content-Security-Policy-Report-Only: default-src 'self'; script-src 'self' https://js.chargebee.com https://js.stripe.com; frame-ancestors 'self'
Content-Type: text/html
Date: Wed, 29 Nov 2023 10:52:41 GMT
Last-Modified: Wed, 29 Nov 2023 05:29:50 GMT
Server: routerapp
Vary: Accept-Encoding
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
---------END-HTTP---------
```

**Review by anjalshireesh — APPROVED** [review]

File: cmd/subnet-utils.go [hunk 1]
```diff
@@ -174,8 +175,63 @@ func getSubnetClient() *http.Client {
...
-	return getSubnetClient().Do(req)
+func subnetHTTPDo(req *http.Request) (resp *http.Response, err error) {
+	resp, err = getSubnetClient().Do(req)
+	if err == nil && globalDebug {
+		dumpHTTPReq(req, resp)
+	}
+	return
+}
+
+// dumpHTTP - dump HTTP request and response.
```
Comment (shtripat) [inline cmd/subnet-utils.go:186]: `%s/dumpHTTP/dumpHTTPReq/g` or actuallythe function itself can be called `dumpHTTP` as it dumps both request and response.

**Review by shtripat — APPROVED** [review]
One minor comment.

File: cmd/subnet-utils.go (hunk 1 above)
Comment (vadmeste) [reply cmd/subnet-utils.go:186]: just saw this; I will fix this in another PR

**Pull request merged**